packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/stackset-types.ts

```diff
@@ -0,0 +1,413 @@
+import * as codepipeline from '@aws-cdk/aws-codepipeline';
+import * as iam from '@aws-cdk/aws-iam';
+import * as cdk from '@aws-cdk/core';
+import { Construct } from 'constructs';
+
+/**
+ * The source of a StackSet template
+ */
+export abstract class StackSetTemplate {
+  /**
+   * Use a file in an artifact as Stack Template.
+   */
+  public static fromArtifactPath(artifactPath: codepipeline.ArtifactPath): StackSetTemplate {
+    return new class extends StackSetTemplate {
+      public readonly artifactsReferenced?: codepipeline.Artifact[] | undefined = [artifactPath.artifact];
+
+      public render() {
+        return artifactPath.location;
+      }
+    }();
+  }
+
+  /**
+   * Which artifacts are referenced by this template
+   *
+   * Does not need to be called by app builders.
+   */
+  public abstract readonly artifactsReferenced?: codepipeline.Artifact[] | undefined;
+
+  /**
+   * Render the template to the pipeline
+   *
+   * Does not need to be called by app builders.
+   */
+  public abstract render(): any;
```

Comment:
Is there seriously no type we can specify here?

If we're going to keep this an `any`, I think thats' an even better reason to make these `@internal`.